### PR TITLE
Circleci: Create install, build, and test jobs for faster builds

### PR DESCRIPTION
**Pull request by mejiaj**
## Description

This PR attempts to improve CircleCI builds to be faster and more reliable. 

⚠️We still need to update to pa11y 3 to avoid those jekyll page crashing issues that have been bothering us.

## Additional information

The previous `build` job is now split into 3 separate "phases". It uses the `persist_to_workspace` and `attach_workspace` features to re-use the same build per workflow.

1. **Install dependencies ⬇️**
   1. Install Ruby deps
   1. Install Node deps 
1. **Build the site 🏗️**
   1. Jekyll Build
   1. Build assets 
1. **Run tests 🧪**
   1. Run ruby unit tests & HTML proofer
   1. Test pa11y desktop
   3. Test pa11y mobile

This changes to workflow so the sub-steps can run at the same time, speeding up builds/rebuilds, and clarity on what exactly failed.

**Before**
image

**After**
image

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [ ] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [ ] Run `npm test` and make sure the tests for the files you have changed have passed.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [ ] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,15 +3,16 @@ version: 2.1
 orbs:
   snyk: snyk/snyk@1.1.2
 
-jobs:
-  build:
+executors:
+  my-executor:
```
Comment (mejiaj) [inline .circleci/config.yml:7]: This is so we don't have to specify the docker image for **each** job. Just saves us 3 lines.

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,15 +3,16 @@ version: 2.1
...
     docker:
       - image: cimg/ruby:3.0.4-browsers
+
+jobs:
+  install_ruby_deps:
+    executor: my-executor
```
Comment (mejiaj) [inline .circleci/config.yml:13]: Our re-used `executor`.

File: .circleci/config.yml [hunk 3]
```diff
@@ -30,24 +31,107 @@ jobs:
           key: gem-cache-{{ checksum "Gemfile.lock" }}
           paths:
             - vendor/bundle
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - "vendor/bundle"
```
Comment (mejiaj) [inline .circleci/config.yml:37]: We're saving the `vendor/bundle` directory for re-use.

File: .circleci/config.yml [hunk 4]
```diff
@@ -30,24 +31,107 @@ jobs:
...
+          command: npm run build-uswds
+      - run:
+          name: Build site assets
+          command: npm run build-all-assets
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - "_site"
+            - "assets"
+            - "node_modules"
```
Comment (mejiaj) [inline .circleci/config.yml:76]: We're saving the site build, the built assets (CSS/JS), and `node_modules` (required for pa11y and USWDS html-templates)

File: .circleci/config.yml [hunk 5]
```diff
@@ -30,24 +31,107 @@ jobs:
...
+            - install_node_deps
+      - test_build:
+          requires:
+            - build
+      - test_a11y_desktop:
+          requires:
+            - build
+      - test_a11y_mobile:
+          requires:
+            - build
```
Comment (mejiaj) [inline .circleci/config.yml:137]: Separated pa11y tests so they can run at the same time

image

Comment (thisisdano) [reply .circleci/config.yml:137]: Hallelujah!

**Review by thisisdano — APPROVED** [review]
What a great improvement — thanks!

**Pull request merged**